### DISCUSSION TITLE
ArgoCD Pre-Sync recreating ConfigMaps

### DISCUSSION THREAD

**Discussion by amits2613** [argoproj/argo-cd#7281] [General]
**Problem Statement:**
Deploying application which require ConfigMap & Secrets need to be present to deploy DB schema migration and application deployment. DB schema migration should finish before actual application deployment.

**Steps taken:**
argocd pre-sync and sync-waves annotations added in ConfigMaps, Secrets & DB Schema Migration job. 

```
  annotations:
    argocd.argoproj.io/hook: PreSync
``` 
**Use-Case:**
In our use-case, we want the ConfigMaps / Secrets to be deployed before deploying actual application. To achieve this, were using ArgoCD Pre-Sync hooks, but now in every of the sync operation the argocd recreating ConfigMaps / Secrets in the pre-sync hook cycle.

We want ConfigMaps/ Secrets to be updated when only there is commit / change in those manifest files, we don't want them to be created every time in the pre-sync.

**Comment by wanghong230** [comment]
Good question. Could you help me understand the side-effect of the no-change update from CM and Secrets from your use case? Kubernetes works in a declarative way, so it will be applied again and again even without changes.

**Reply by amits2613** [reply]
ArgoCD not updating CM / Secrets but actually deleting those first and then recreate it. 

There will be issue in deletion and then re-creation because if our cronjob gets scheduled during that time difference , it will fail as it wont be able to find that configmap .

Moreover, if deleting and then recreating we will loss the track of changes done in the configmap
